plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/procedure/MigrateProcedure.java

```diff
@@ -307,12 +309,23 @@ private Schema toIcebergSchema(List<Column> columns)
         return new Schema(icebergSchema.asStructType().fields());
     }
 
-    private static org.apache.iceberg.types.Type toIcebergType(Type type, AtomicInteger nextFieldId)
+    private static org.apache.iceberg.types.Type toIcebergType(Type type, AtomicInteger nextFieldId, HiveStorageFormat storageFormat)
     {
         if (type instanceof ArrayType || type instanceof MapType || type instanceof RowType) {
             // TODO https://github.com/trinodb/trino/issues/17583 Add support for these complex types
             throw new TrinoException(NOT_SUPPORTED, "Migrating %s type is not supported".formatted(type));
         }
+
+        // TODO: Add support for structural types with timestamp(3)
+        if (type.equals(TimestampType.TIMESTAMP_MILLIS)) {
+            if (storageFormat == HiveStorageFormat.ORC || storageFormat == HiveStorageFormat.PARQUET) {
+                return Types.TimestampType.withoutZone();
+            }
+            // Adding support for Avro is currently suspended as there parallel works on switching to Avro native reader.
+            if (storageFormat == HiveStorageFormat.AVRO) {
+                throw new TrinoException(NOT_SUPPORTED, format("Timestamp precision (%s) not supported for Iceberg with Avro file format. Use \"timestamp(6)\" instead.", TimestampType.TIMESTAMP_MILLIS.getPrecision()));
+            }
+        }
```

Comment:
nit: We can use switch statement so that we can avoid updating this method when Iceberg adds new file format. 
```java
        if (type.equals(TimestampType.TIMESTAMP_MILLIS)) {
            return switch (storageFormat) {
                case ORC, PARQUET -> Types.TimestampType.withoutZone();
                default -> throw new TrinoException(NOT_SUPPORTED, "Migrating timestamp(3) type with %s format is not supported. Use \"timestamp(6)\" instead.".formatted(storageFormat));
            };
        }
```